packages/plugins/apq/__tests__/automatic-persisted-queries.spec.ts

```diff
@@ -57,6 +58,7 @@ describe('Automatic Persisted Queries', () => {
     const response = await request(yoga)
       .post('/graphql')
       .send({
+        query: '',
```

Comment:
I don't think clients supporting APQ provide `query`. If they don't, this would break the support of those clients.

Reply:
[`query` is a required parameter.](https://graphql.github.io/graphql-over-http/draft/#sel-EALFFEBAAACDAx_L)

Reply:
Yes but this is a different spec we follow. We are going beyond spec here and we want to support clients instead of introducing our own.

Reply:
Note to be spec compliant query must not only be present but also must be the string representation of the GraphQL document. An empty string does not satisfy the spec (and is weird/confusing).

Persisted operations are deliberately not covered by the GraphQL-over-HTTP spec right now, this is because the spec is to increase interoperability but persisted operations (when used as an allowlist) have exactly the opposite aim. That’s fine - you can use persisted operations alongside a GraphQL compliant server, it’s just an alternative communication method (not spec compliant, but still perfectly fine). We’ll probably incorporate some wording for persisted operations in a post-1.0 release of the spec.